Contact driver

hanc2006 opened the pull request:
Added missing TelegramContactDriver extension from TelegramDriver base class

Commit:
added contact driver

Commit:
removed self-generated files

hanc2006 committed:
added contact driver (#52)

* added contact driver

* removed self-generated files

Commit:
extended TelegramDriver with Contact

Commit:
added Contact namespace

holtkamp commented:
I guess this can / should be merged so a PR can be sent to https://github.com/botman/driver-telegram ?

sh41 committed:
Fix bug in sending Contact and update tests to test both with and without a vcard.

sh41 committed:
Cs Fixer

sh41 committed:
The documentation for the free styleCI config is here: https://docs.styleci.io/standalone-php linter is not mentioned. Removed to fix
```
Config Issue
The provided config option 'linting' was not valid.
```

sh41 committed:
Add configuration option for throw_http_exceptions to ensure that there are no BC breaks.

When set to true this will perform various checks on the response from an http call and throw a TelegramConnectionException if there are any problems. The message content will contain debugging information.

Have added new tests for this.

Also fixed it_throws_exception_in_get_user test as it would not have correctly picked up situation where no exception was thrown.

sh41 committed:
Filter telegram token from exception message. Prevent a credential leak!

sh41 committed:
Add option to retry for server failures and 429 rate warnings.

sh41 committed:
Change .styleci to match PSR2 as stated in contributing guidelines, and update files to PSR2 rules.

feralheart committed:
Merge pull request #73 from sh41/feature/http_error_checking

Optionally throw exception when http call fails.

sh41 committed:
Fix bug and failing tests in sendContact (#71)

* Fix bug in sending Contact and update tests to test both with and without a vcard.

* Cs Fixer

Co-authored-by: Steve Hall <steve.hall+gitlab@rg456.co.uk>

sh41 committed:
Add configuration option to enable setting of default additional options to be sent with every request. Specifically this is to enable defaulting all messages to Markdown format by having the following config: (#70)

```
'telegram' => [
	'token' => 'YOUR-TELEGRAM-TOKEN-HERE',
	'default_additional_parameters' =>
	    [
	        'parse_mode' => 'Markdown',
	    ],
]
```

Co-authored-by: Steve Hall <steve.hall+gitlab@rg456.co.uk>
Co-authored-by: FeralHeart <feralheart@feralsystems.com>

antimech committed:
Create ISSUE_TEMPLATE.md (#59)

Co-authored-by: antimech <antimech@users.noreply.github.com>

antimech committed:
Fix repo link in CONTRIBUTING.md (#57)

Co-authored-by: antimech <antimech@users.noreply.github.com>

feralheart committed:
Add channel_post to buildPayload (#69)

antimech committed:
Update book link (#58)

Use https instead of http.

Co-authored-by: antimech <antimech@users.noreply.github.com>

Commit:
Don't remove inline keyboard automatically (Closes #44, closes #29, also closes botman/botman#709) (#48)

feralheart committed:
Added Hall-of-fame to README (#78)

feralheart committed:
Merge branch 'master' into contact-driver
